### PR TITLE
Fix tuple metavar crash in Cmd2HelpFormatter and ArgparseCompleter

### LINKED ISSUE #974: Bug in Cmd2HelpFormatter and ArgparseCompleter when metavar is a tuple
The [Cmd2HelpFormatter](https://github.com/python-cmd2/cmd2/blob/master/cmd2/argparse_custom.py#L600-L716) class has a bug which causes a crash in the following circumstances:
- `nargs` is an int and > 1
- [metavar](https://docs.python.org/3/library/argparse.html#metavar) is a tuple

Tested with `cmd2` 1.2.0 but verified the bug exists in the current version as well.

In my case, the crash occurs on line 780 in the current code with the [_format_args](https://github.com/python-cmd2/cmd2/blob/master/cmd2/argparse_custom.py#L763-L784) method.

### PR DESCRIPTION
**Pull request by kmvanbrunt**
Fixes #974

### CODE REVIEW

File: cmd2/argparse_custom.py [hunk 1]
```diff
@@ -743,38 +744,46 @@ def _metavar_formatter(self, action, default_metavar) -> Callable:
...
+                return (metavar, ) * tuple_size
         return format
 
     # noinspection PyProtectedMember
     def _format_args(self, action, default_metavar) -> str:
-        get_metavar = self._metavar_formatter(action, default_metavar)
-        # Begin cmd2 customization (less verbose)
-        nargs_range = getattr(action, ATTR_NARGS_RANGE, None)
+        """Customized to handle ranged nargs and make other output less verbose"""
+        metavar = self._determine_metavar(action, default_metavar)
```
Comment (anselor) [inline cmd2/argparse_custom.py:763]: We're calling `self._determine_metavar()` twice in a row here. Maybe pass it into `self._metavar_formatter()` or have `self._metavar_formatter()` return a tuple?

File: cmd2/argparse_custom.py [hunk 2]
```diff
@@ -743,38 +744,46 @@ def _metavar_formatter(self, action, default_metavar) -> Callable:
             # End cmd2 customization
         else:
             result = default_metavar
+        return result
+
+    def _metavar_formatter(self, action, default_metavar) -> Callable:
```
Comment (anselor) [inline cmd2/argparse_custom.py:749]: The name makes me think the callable returned here is the output formatter, but it's just trying to get the data into a standard data structure. Maybe call something like `_normalize_metavar()`?

File: cmd2/argparse_custom.py [hunk 3]
```diff
@@ -743,38 +744,46 @@ def _metavar_formatter(self, action, default_metavar) -> Callable:
...
+
+        # Make this output less verbose. Do not customize the output when metavar is a
+        # tuple of strings. Allow argparse's formatter to handle that instead.
+        elif isinstance(metavar, str):
+            if action.nargs == ZERO_OR_MORE:
+                return '[%s [...]]' % metavar_formatter(1)
+            elif action.nargs == ONE_OR_MORE:
+                return '%s [...]' % metavar_formatter(1)
+            elif isinstance(action.nargs, int) and action.nargs > 1:
+                return '{}{{{}}}'.format('%s' % metavar_formatter(1), action.nargs)
```
Comment (anselor) [inline cmd2/argparse_custom.py:784]: Every call to `metavar_formatter()` passes in 1. Is something supposed to pass in something other than 1? If not, do we need that function argument?

File: cmd2/argparse_custom.py (hunk 1 above)
Comment (kmvanbrunt) [reply cmd2/argparse_custom.py:763]: `self._metavar_formatter()` is an overrided `argparse` function. I can't change its signature.

File: cmd2/argparse_custom.py (hunk 3 above)
Comment (kmvanbrunt) [reply cmd2/argparse_custom.py:784]: `self._metavar_formatter()` is an overrided `argparse` function. It expects the size of tuple being created as an argument.

File: cmd2/argparse_custom.py (hunk 2 above)
Comment (kmvanbrunt) [reply cmd2/argparse_custom.py:749]: I can't rename this since it's an overrided `argparse` method.

**Comment by anselor** [comment]
Looks good.

**Pull request merged**